### ISSUE TITLE
Cypress logs timeout, fails test, but keeps running all commands. no way to know why it is timeing out. 

### ISSUE THREAD

**Issue by egucciar** [cypress-io/cypress#2889]
### Current behavior:

Cypress "randomly" is logging a Timeout failure but is not linking it to anything specific. These commands typically coincide with my custom commands, which are not timeouting before the specified timeout in other projects and when they do, it throws the appropriate message. 

I've already ruled out the custom command as an issue because it works as expected in other projects. Rather, I want to **debug cypress** but finding it to be **impossible** given the current lack of information provided by the runner.

This is the end of the tests (All my commands and assertions ran correctly)

![image](https://user-images.githubusercontent.com/2848185/49400520-87865580-f712-11e8-9ff1-bf7319fa4356.png)

It already "errored" out on command One, but it kept going until this moment. Why did it keep running commands if it failed? 

Clicking the error yields no additional information:

![image](https://user-images.githubusercontent.com/2848185/49400572-a553ba80-f712-11e8-91d4-8606a079023d.png)

Adding a debug handler for the failure yields nothing (it doesnt stopp here unless one of my assertions actually does fail)

```js
Cypress.on("fail", (err, runnable) => {
  debugger
  return err
})
```

### Desired behavior:

I want to be able to figure out the root cause of this timeout.

### Steps to reproduce:

None so far; this is only happening in one test / one project, despite us using these commands practically everywhere.

Cypress v 3.1.0 and up

**Comment by egucciar** [comment]
By searching through this code, I discovered many or most commands make a call to `cy.clearTimeout('commandName')`. 

Though I do not know why, and why other tests are not failing though obviously this "should" be a "global" problem for all of my async custom commands, adding the `cy.clearTimeout` prevents the failure from occuring:

![image](https://user-images.githubusercontent.com/2848185/49403026-9f150c80-f719-11e8-9f2f-a6ec3a276532.png)

**Comment by egucciar** [comment]
Another note: these failures did NOT occur in CI (local open mode, only) and do NOT occur for all team members. My custom commands work as expected with even large timeoutd of 30+ seconds across many projects and hundreds of tests so there's no clue as to why this happened in one environment in one test

**Comment by jennifer-shehane** [comment]
When attempting to catch an error, you'll want to `throw err`

```js
Cypress.on("fail", (err, runnable) => {
  debugger
  throw err
})
```

Also, I'd recommend adding some log events to your custom command to get some visibility in the Command Log about where you are within your custom command. Even something as simple as `cy.log('command name')` or [`Cypress.log()`](https://on.cypress.io/cypress-log)

**Comment by egucciar** [comment]
@jennifer-shehane unfortunately, the debugger doesn't get hit unless my test code fails. This timeout does not ultimately call into this handler so it's moot point since I'm trying to Target the failure, but that failure isn't getting here. (Notable; other valid timeout failures are). 

Also, is this log for debugging purpose? If so are we trying to track down between what steps the timeout fail occurs? As I was able to directly breakpoint into cypress code I can see if there's a correlation but since this is always happening after 4 s, I'm assuming it's in the getElements / retryability loop since my elements take about 6s to load on average. But, in any case I think that the failure itself correlated not with any specific command but the test just failing in general. It seems like to me there's a timeout that's just being set , automatically, and cleared every so often by commands as they run. At least, it did look that way. If that hypothesis is correct I should be able to remove the wait time for the elements and observe if the test fails even if my gets all returned syncronously.

**Comment by egucciar** [comment]
So far I've tried replicating "working" test from one project into the other. The working project doesn't fail for this reason. This project does. It also impacted another Dev. I think I know why it might not be happening on CI as well .

So I've ruled out, different setting, different preprocessors, different dependency versions , and no luck. This is definitely strange. Going to keep trying some other things. Will probably see if I can reproduce the issue with out my app code and just the cypress setup and custom commands. Sadly I just have no time to look into that this week. I'll probably keep my clear fix around for bow and get back to this when I can.
